### PR TITLE
py3 userreports filters

### PR DESCRIPTION
**Pull request by nickpell**
@emord @millerdev

### CODE REVIEW

File: corehq/apps/userreports/filters/__init__.py [hunk 1]
```diff
@@ -30,7 +32,7 @@ def __init__(self, filters):
         assert len(self.filters) > 0
 
     def __call__(self, item, context=None):
-        return all(filter(item, context) for filter in self.filters)
+        return all(list(filter(item, context)) for filter in self.filters)
```
Comment (millerdev) [inline corehq/apps/userreports/filters/__init__.py]: This doesn't look right. `filter` is a local here, not the global/builtin `filter`. Does it return a list?

File: corehq/apps/userreports/filters/__init__.py [hunk 2]
```diff
@@ -43,7 +45,7 @@ def __init__(self, filters):
         assert len(self.filters) > 0
 
     def __call__(self, item, context=None):
-        return any(filter(item, context) for filter in self.filters)
+        return any(list(filter(item, context)) for filter in self.filters)
```
Comment (millerdev) [inline corehq/apps/userreports/filters/__init__.py]: Same as above

**Review by millerdev — CHANGES_REQUESTED** [review]
Test failure is real:
```
ERROR: test_complex_structure (corehq.apps.userreports.tests.test_filters.ComplexFilterTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_filters.py", line 359, in test_complex_structure
    self.assertTrue(filter(dict(foo='bar')))
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/filters/__init__.py", line 48, in __call__
    return any(list(filter(item, context)) for filter in self.filters)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/filters/__init__.py", line 48, in <genexpr>
    return any(list(filter(item, context)) for filter in self.filters)
TypeError: 'bool' object is not iterable
```

File: corehq/apps/userreports/filters/__init__.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/userreports/filters/__init__.py]: nice catch

**Review by millerdev — APPROVED** [review]

**Pull request merged**